### PR TITLE
View Followers and Following Lists

### LINKED ISSUE #141: Implement follower/following list
**User statement**
*As a user, I would like to see who I'm following and who my followers are.

**Describe the solution you'd like**
Implement a couple of new fragments to view the follower/following lists.

**Development notes**

### PR DESCRIPTION
**Pull request by kaden-dreger**
This PR deals with the creation of the Following and Followers lists inside the Profile Page. The user can click on the number of followers/following button which will perform a navigation action to the new FollowListFragment. In this fragment you can see all of your followers/following in a list with the username and profile picture associated with those present. This PR does not add any unit tests or UI tests. The UI tests will be added in the story that deals with adding the unfollow button functionality: #171. The profile pictures are currently not implemented. This PR closes #141.

- [x] run unit tests
- [x] run ui tests

Profile page
Screen Shot 2021-11-25 at 7 05 38 PM
:

Followers page:
Screen Shot 2021-11-25 at 7 07 22 PM

Following page:
Screen Shot 2021-11-25 at 7 07 13 PM

Swipe for unfollow:
Screen Shot 2021-11-25 at 7 07 32 PM

### CODE REVIEW

File: app/src/main/java/com/example/habitsmasher/ui/profile/FollowListFragment.java
Comment (kaden-dreger) [inline app/src/main/java/com/example/habitsmasher/ui/profile/FollowListFragment.java:221]: Currently just displays a toast message, this will be replaced with story #171

File: app/src/main/java/com/example/habitsmasher/ui/profile/FollowListFragment.java
Comment (kaden-dreger) [inline app/src/main/java/com/example/habitsmasher/ui/profile/FollowListFragment.java:229]: Currently just displays a toast message, this will be replaced with story #22 .

File: app/src/main/java/com/example/habitsmasher/ui/profile/ProfileFragment.java
Comment (kaden-dreger) [inline app/src/main/java/com/example/habitsmasher/ui/profile/ProfileFragment.java:63]: Changed the TextViews to Buttons, this can be undone and just have the text clickable. I think that this makes it very obvious that it is clickable however.

File: app/src/main/java/com/example/habitsmasher/ui/profile/FollowListFragment.java
Comment (rudy-patel) [inline app/src/main/java/com/example/habitsmasher/ui/profile/FollowListFragment.java:225]: minor: typo in **function**

File: app/src/main/java/com/example/habitsmasher/ui/profile/ProfileFragment.java
Comment (rudy-patel) [inline app/src/main/java/com/example/habitsmasher/ui/profile/ProfileFragment.java:55]: - [ ] extract to constants

File: app/src/main/java/com/example/habitsmasher/ui/profile/ProfileFragment.java
Comment (rudy-patel) [reply app/src/main/java/com/example/habitsmasher/ui/profile/ProfileFragment.java:63]: I think I'd prefer if the buttons were white, with a slight outline instead of them being jarring blue on an all-white page. Just my personal preference.

File: app/src/test/java/com/example/habitsmasher/UserAccountHelperTest.java
Comment (rudy-patel) [inline app/src/test/java/com/example/habitsmasher/UserAccountHelperTest.java:22]: - [ ] extract to constants

File: app/src/test/java/com/example/habitsmasher/test/UserTest.java
Comment (rudy-patel) [inline app/src/test/java/com/example/habitsmasher/test/UserTest.java:17]: - [ ] extract to constants

**Review by rudy-patel — CHANGES_REQUESTED** [review]
Left a few tasks regarding things that need clean up. Also, I'd consider changing the colour of the follower/following count buttons to be white with a slight outline (so they know it's still a button, but way less visually jarring on an all-white page).

Once the tasks are implemented, you have my approval.

File: app/src/main/java/com/example/habitsmasher/User.java
Comment (Nova-Drive) [reply app/src/main/java/com/example/habitsmasher/User.java:54]: It's for the recyclerview builder, for the list of followers and following. Same reason habits have the tag.

**Review by Nova-Drive — COMMENTED** [review]
Nice job! I'll approve once you finish Rudy's review.

File: app/src/main/java/com/example/habitsmasher/ui/profile/FollowListFragment.java
Comment (kaden-dreger) [reply app/src/main/java/com/example/habitsmasher/ui/profile/FollowListFragment.java:132]: For now this is a working solution and its being used in other areas of the project. For now lets move forward with this.

File: app/src/main/java/com/example/habitsmasher/ui/profile/FollowItemAdapter.java
Comment (kaden-dreger) [reply app/src/main/java/com/example/habitsmasher/ui/profile/FollowItemAdapter.java:57]: This will be implemented with story: #167

**Review by rudy-patel — DISMISSED** [review]

**Review by rudy-patel — APPROVED** [review]

**Review by jaskim9824 — APPROVED** [review]

**Review by Nova-Drive — DISMISSED** [review]

**Review by rudy-patel — APPROVED** [review]

**Review by Nova-Drive — APPROVED** [review]

**Pull request merged**
